receiver/k8seventsreceiver/receiver.go

```diff
@@ -50,15 +56,26 @@ func newReceiver(
 	return &k8seventsReceiver{
 		settings:     set,
 		config:       config,
+		options:      options,
 		logsConsumer: consumer,
 		startTime:    time.Now(),
 		obsrecv:      obsrecv,
 	}, nil
 }
 
-func (kr *k8seventsReceiver) Start(ctx context.Context, _ component.Host) error {
+func (kr *k8seventsReceiver) Start(ctx context.Context, host component.Host) error {
 	kr.ctx, kr.cancel = context.WithCancel(ctx)
 
+	allOptions := append(createReceiverOpts(kr.config), kr.options...)
+
+	for _, opt := range allOptions {
+		if err := opt(kr); err != nil {
+			kr.settings.Logger.Error("Could not apply option", zap.Error(err))
+			componentstatus.ReportStatus(host, componentstatus.NewFatalErrorEvent(err))
```

Comment:
Should this be a fatal error?

Reply:
I believe that yes, as the incorrect configuration shouldn't be executed as it's hard to catch later just from logs error reporting, better to fail immediately